### DISCUSSION TITLE
How to refresh PATH without reinstalling?

### DISCUSSION THREAD

**Discussion by JulienTant** [d12frosted/homebrew-emacs-plus#701] [General]
### Feature request description

Do you know of a way to refresh the injected `PATH` without reinstalling?

Every time I have to update the `PATH` my next step has been to reinstall but I wonder if there's a quicker way to make this happen? Sometimes it's just not convenient to have to wait 😅 

If the best solution is to install `[exec-path-from-shell](https://github.com/purcell/exec-path-from-shell)` I'm cool with this ~ just wondered if there's something "provided" with the formulae to make this happen without needed the extensions or recompiling!

**Comment by neoheartbeats** [comment]
You may check the variable `exec-path`, for example, the following appends `.emacs.d/bin/` to PATH:

```
(add-to-list 'exec-path (expand-file-name "bin/" user-emacs-directory))
```

**Comment by MatthiasPortzel** [comment]
The app uses the path from `$(find $(brew --prefix) -name 'Emacs.app')/Contents/Info.plist`. You could edit the `PATH` string in that file, but it will be overwritten on reinstall or update.

**Comment by d12frosted** [comment]
The PATH injection feature is mostly for non-dynamic environments. You have two options. Either you modify `Info.plist` (i.e. what path injection feature does) - basically what @MatthiasPortzel does. Or you modify env from within Emacs either manually or using `exec-path-from-shell` package. 

BTW, remember that environment is inherited if you start application from shell. Say, you are using terminal, you modify PATH and then start `emacs -nw` or just `emacs` - both ways you inherit the modified PATH. It's useful in some flows.
